modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,506 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains class ModinDataframe.
+
+ModinDataframe is a parent abstract class for any dataframe class.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Hashable, Optional, Callable, Union, Dict
+
+
+class ModinDataframe(ABC):
+    """
+    An abstract class that represents the Parent class for any DataFrame class.
+
+    This class is intended to specify the behaviours that a DataFrame must implement.
+
+    For more details about how these methods were chosen, please refer to this
+    (http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf) paper, which specifies
```

Comment:
> we should probably add a link to a newer paper

or better link to a static section on read-the-docs that always holds a link to the newest/actual paper

Reply:
Should we open a new pr to creat this section in the docs, or should I just add a link to the new paper here?

Reply:
We already have a link to http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf in the readme.md file that should be updated but I would prefer to add a section in the docs where we give a recap of the newer paper and provide a link to it. Of course, as part of a separate issue.

Reply:
#3858

Reply:
Since the changes in this PR reflect theses from https://people.eecs.berkeley.edu/~totemtang/paper/Modin.pdf I suggest updating the link to that paper.